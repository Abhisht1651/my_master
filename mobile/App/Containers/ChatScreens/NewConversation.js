/* eslint-disable react/jsx-props-no-spreading */
// @flow
import React, { useState } from 'react'
import { View, Text, TouchableOpacity, ScrollView, FlatList, Image } from 'react-native'
import { NavigationContainerProps } from '@react-navigation/native'
import SearchBarDefault from '../../Components/Searchbars/SearchBarDefault'
import Images from '../../Themes/Images'
import { LargeThinButton } from '../../Components/Buttons/LargeThinButton'

// Styles
import styles from './ScreenStyles/NewConversationStyles'
import NewConversationList from '../../Components/NewConversationList/NewConversationList'

type Props = NavigationContainerProps & {
  item: {
    name: string,
    source: string
  }
}
const Data = [
  {
    id: '1',
    name: 'Amelia Jones',
    source: Images.profile2
  },

  {
    id: '2',
    name: 'John Doe',
    source: Images.profilePicture3
  },

  {
    id: '3',
    name: 'Jim Moriarty',
    source: Images.profilePicture4
  },

  {
    id: '4',
    name: 'Ash Ketchum',
    source: Images.profilePicture
  },

  {
    id: '5',
    name: 'Richard Hendricks',
    source: Images.profile1
  },

  {
    id: '6',
    name: 'Carry Minati',
    source: Images.profilePicture2
  },

  {
    id: '7',
    name: 'John Snow',
    source: Images.profileCamera
  },

  {
    id: '8',
    name: 'Dwanye Johnson',
    source: Images.profilePicture1
  },

  {
    id: '9',
    name: 'James Anderson',
    source: Images.profilePicture3
  },

  {
    id: '10',
    name: 'Robin Van Persie',
    source: Images.profilePicture4
  }
]

const NewConversation = (props: Props) => {
  const { navigation } = props
  const renderfriends = ({ item }: Props) => {
    if (onSearchFilter(item.name) === true || search === '') {
      return <NewConversationList name={item.name} image={item.source} />
    }
    return null
  }
  const [search, setsearch] = useState('')

  const onSearch = (value) => {
    setsearch(value)
  }

  const onSearchFilter = (name) => {
    const name_arr = name.split(' ')
    const searched = search.toUpperCase()
    const search_len = searched.length
    for (let i = 0; i < name_arr.length; i += 1) {
      if (search_len <= name_arr[i].length) {
        const sub_str = name_arr[i].substring(0, search_len)
        if (sub_str.toUpperCase() === searched) {
          return true
        }
      }
    }
    if (name.length >= search_len) {
      const sub_str = name.substring(0, search_len)
      if (sub_str.toUpperCase() === searched) {
        return true
      }
    }
    return false
  }
  return (
    <View style={styles.screen}>
      <View style={styles.topBox}>
        <View style={styles.topBox1}>
          <View style={styles.backbutton}>
            <TouchableOpacity onPress={() => navigation.navigate('MessageScreen')}>
              <Image source={Images.arrowleft} style={styles.arrowleft} />
            </TouchableOpacity>
          </View>
          <Text style={styles.titleFont}>New Conversation</Text>
          <TouchableOpacity onPress={() => navigation.navigate('InviteFriends')}>
            <Text style={styles.importFont}>Invite</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.topBox2}>
          <SearchBarDefault onSearchHandler={onSearch} />
        </View>
      </View>
      <View style={styles.mainBox}>
        <View style={styles.listBox}>
          <ScrollView contentContainerStyle={styles.listAlign}>
            <View style={styles.list}>
              <FlatList
                data={Data}
                renderItem={renderfriends}
                showsVerticalScrollIndicator={false}
                style={styles.listElements}
              />
            </View>
          </ScrollView>
        </View>
      </View>
      <View style={styles.bottomBox}>
        <LargeThinButton title="Start" />
      </View>
    </View>
  )
}

export default NewConversation
