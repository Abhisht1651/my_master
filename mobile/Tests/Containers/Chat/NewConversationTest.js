import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity, Text, Image, FlatList, ScrollView } from 'react-native'
import NewConversation from '../../../App/Containers/ChatScreens/NewConversation'

describe('NewConversation', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<NewConversation />) // no compile-time error
  })
  it('NewConversation component renders correctly', () => {
    const wrapper = shallow(<NewConversation />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 2 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(2)
  })
  it('should render 2 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(2)
  })
  it('should render 1 Image', () => {
    expect(wrapper.find(Image)).toHaveLength(1)
  })
  it('should render 1 Flatlist', () => {
    expect(wrapper.find(FlatList)).toHaveLength(1)
  })
  it('should render 1 ScrollView', () => {
    expect(wrapper.find(ScrollView)).toHaveLength(1)
  })
})
