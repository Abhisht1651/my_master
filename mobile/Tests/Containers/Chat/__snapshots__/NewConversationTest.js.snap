// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewConversation NewConversation component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": "column",
      "height": 1334,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#E6ECED",
        "borderBottomWidth": 2,
        "elevation": 2,
        "height": 192.2142857142857,
        "paddingHorizontal": 48,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 101.85714285714286,
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "height": 34,
                "width": 44,
              }
            }
          />
        </ForwardRef>
      </View>
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 40,
            "letterSpacing": -0.28,
          }
        }
      >
        New Conversation
      </Text>
      <ForwardRef
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#7B46E4",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 32,
              "letterSpacing": -0.18,
            }
          }
        >
          Invite
        </Text>
      </ForwardRef>
    </View>
    <View
      style={
        Object {
          "height": 73.92857142857143,
          "width": "100%",
        }
      }
    >
      <SearchBarDefault
        onSearchHandler={[Function]}
        onSubmitEditing={[Function]}
        onTouchStart={[Function]}
        placeholder="Search"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 50,
        "paddingTop": 55.857142857142854,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 944.6428571428571,
          "width": "100%",
        }
      }
    >
      <ScrollViewMock>
        <View>
          <FlatList
            data={
              Array [
                Object {
                  "id": "1",
                  "name": "Amelia Jones",
                  "source": Object {
                    "default": "",
                  },
                },
                Object {
                  "id": "2",
                  "name": "John Doe",
                  "source": Object {
                    "default": "",
                  },
                },
                Object {
                  "id": "3",
                  "name": "Jim Moriarty",
                  "source": Object {
                    "default": "",
                  },
                },
                Object {
                  "id": "4",
                  "name": "Ash Ketchum",
                  "source": Object {
                    "default": "",
                  },
                },
                Object {
                  "id": "5",
                  "name": "Richard Hendricks",
                  "source": Object {
                    "default": "",
                  },
                },
                Object {
                  "id": "6",
                  "name": "Carry Minati",
                  "source": Object {
                    "default": "",
                  },
                },
                Object {
                  "id": "7",
                  "name": "John Snow",
                  "source": Object {
                    "default": "",
                  },
                },
                Object {
                  "id": "8",
                  "name": "Dwanye Johnson",
                  "source": Object {
                    "default": "",
                  },
                },
                Object {
                  "id": "9",
                  "name": "James Anderson",
                  "source": Object {
                    "default": "",
                  },
                },
                Object {
                  "id": "10",
                  "name": "Robin Van Persie",
                  "source": Object {
                    "default": "",
                  },
                },
              ]
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsVerticalScrollIndicator={false}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </View>
      </ScrollViewMock>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 180.71428571428572,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <LargeThinButton
      disabled={false}
      onPress={[Function]}
      other={false}
      outline={false}
      style={Object {}}
      title="Start"
    />
  </View>
</View>
`;
